### PR TITLE
Update Chinese document for max_query_size, max_parser_depth and optimize_functions_to_subcolumns

### PR DESCRIPTION
**Pull request by Alex-Cheng**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Update Chinese document for max_query_size, max_parser_depth and optimize_functions_to_subcolumns. Meanwhile add some missing content in the referenced Chinese document nullable.md and array.md.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

### CODE REVIEW

File: docs/zh/sql-reference/data-types/array.md [hunk 1]
```diff
@@ -66,3 +66,27 @@ SELECT array(1, 'a')
 Received exception from server (version 1.1.54388):
 Code: 386. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: There is no supertype for types UInt8, String because some of them are String/FixedString and some of them are not.
 ```
+
+## 数组大小 {#array-size}
+
+可以使用`size0`子列找到数组的大小，而无需读取整个列。对于多维数组，您可以使用`sizeN-1`，其中`N`是所需的维度。
```
Comment (ucasfl) [inline docs/zh/sql-reference/data-types/array.md:72]: 中英文以及中文和数字之间可以前后留一个空格。

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit f3a700b9768a65b4e2e94a71905508b1d50a7614 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/60541/f3a700b9768a65b4e2e94a71905508b1d50a7614/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: docs/zh/sql-reference/data-types/array.md (hunk 1 above)
Comment (Alex-Cheng) [reply docs/zh/sql-reference/data-types/array.md:72]: `size0`这个是一个列名（identifier），不能写成`size 0`。 `sizeN-1`也是一样，例如N=3时，这个应该是`size2`，中间不能有空格。

我参考了英文文档。https://github.com/ClickHouse/ClickHouse/blob/master/docs/en/sql-reference/data-types/array.md#array-size

Comment (ucasfl) [reply docs/zh/sql-reference/data-types/array.md:72]: 我说的是中文和英文之间
```suggestion
可以使用 `size0` 子列找到数组的大小，而无需读取整个列。对于多维数组，您可以使用 `sizeN-1`，其中 `N` 是所需的维度。
```

Comment (Alex-Cheng) [reply docs/zh/sql-reference/data-types/array.md]: 为什么要中英文之间要留空格啊？是习惯性约定吗？

Comment (Alex-Cheng) [reply docs/zh/sql-reference/data-types/array.md]: 谢谢。我其他地方也改了一下。数字和中文标点之间要不要空格，比如这个片段：
> 则返回 `1`，否
要不要 1 和 逗号之间也加空格。变成：
> 则返回 `1` ，否

另外我看到中文文档的其他地方没有按照这个来，中文文档可能是因为历史原因格式还有标注都有点乱。

Comment (ucasfl) [reply docs/zh/sql-reference/data-types/array.md]: 数字和标点之间不需要

**Review by ucasfl — APPROVED** [review]

**Comment by ucasfl** [comment]
```
Error: Docs markdown link couldn't be resolved: (../../sql-reference/functions/tuple-map-functions.md) in "/opt/clickhouse-docs/docs/zh/operations/settings/settings.md" for version current
    at Array.forEach (<anonymous>)
```

**Comment by Alex-Cheng** [comment]
> rkdown link couldn't be resolved:

谢谢，我看看怎么回事。。。

**Comment by Alex-Cheng** [comment]
> ```
> Error: Docs markdown link couldn't be resolved: (../../sql-reference/functions/tuple-map-functions.md) in "/opt/clickhouse-docs/docs/zh/operations/settings/settings.md" for version current
>     at Array.forEach (<anonymous>)
> ```
好像是中文的对应的文档路径是这个 `/docs/zh/sql-reference/functions/tuple-map-functions.mdx`。我改过来了。

**Comment by Alex-Cheng** [comment]
I don't know why the following errors happened. @ucasfl could you please help me have a look at it?
```
Error:  Loading of version failed for version current
Error:  Unable to build website for locale en.
Error:  Error: Invalid sidebar file at "sidebars.js".
```

**Comment by Alex-Cheng** [comment]
@Algunenano Please merge the PR if there are no problems about the it. Thank you.

**Pull request merged**